prov/util/src/util_mem_monitor.c

```diff
@@ -218,9 +222,7 @@ int ofi_monitors_add_cache(struct ofi_mem_monitor **monitors,
 
 		if (dlist_empty(&monitor->list)) {
 			ret = monitor->start(monitor);
-			if (ret == -FI_ENOSYS)
-				continue;
-			else if (ret)
+			if (ret)
 				goto err;
```

Comment:
>if a system memory monitor failed to start, the error was ignored.
As a result, the MR cache was initialized successfully, but it
does not support system memory.

If system memory monitor failed to start, `success_count` won't be incremented, wondering why the MR cache was initialized successfully? (because the followup device memory monitor is started successfully?)

NIT: minitor->monitor in commit message

Reply:
success_count can increase when Provider specified multiple monitors. For example, EFA specify two monitors: one for system, the other for cuda. Even though system monitor failed to start, cuda monitor started successfully, so success_count would be increased.

Reply:
Thanks for the confirmation, @wzamazon. 
Given this is a cherry pick PR, not sure what's the proper way to fix the typos in the commit message of 3c74b53.

Reply:
I will amend just the commit message to fix the typo and update the PR.